libs/Microsoft.MixedReality.WebRTC.Native/src/interop/global_factory.h

```diff
@@ -16,24 +16,59 @@ enum class ObjectType : int {
 
 /// Global factory wrapper adding thread safety to all global objects, including
 /// the peer connection factory, and on UWP the so-called "WebRTC factory".
+/// Usage pattern:
+///   - Get a RefPtr<> to the singleton instance using |InstancePtr()|. This may
+///   initialize a new instance. This also adds a temporary reference to the
+///   factory to keep it alive until the RefPtr<> goes out of scope.
+///   - Use the factory, for example to get the peer connection factory or to
+///   add/remove some wrapper objects. Whether or not any operation fails, the
+///   factory cannot be destroyed due to the temporary reference, even if no
+///   alive object is present anymore.
+///   - When the RefPtr<> goes out of scope, it releases its temporary reference
+///   to the factory. If that reference is the last one, then the factory
+///   attempts to shutdown, that is goes to check the alive objects, which
+///   constitutes long-term references and will also prevent its destruction.
+///
+/// The delayed call to |TryShutdown()| from releasing the temporary reference
+/// when RefPtr<GlobalFactory> goes out of scope instead of when no object is
+/// alive anymore allows the caller to create a wrapper object and fail
+/// initializing it, and remove it from the global factory but without the
+/// factory destroying itself yet. This works around a problem where the object
+/// removing itself triggers the destruction of the global factory singleton
+/// instance while another thread already acquired a reference to the global
+/// factory but did not yet add any wrapper object so cannot prevent its
+/// destruction, and is left with a dangling reference to a destroyed singleton.

```

Comment:
nit: I'd avoid mentioning an issue that cannot happen anymore 😄 Maybe move this to the commit/PR message?

Reply:
Rephrased into user instructions, is that better? There is a strict usage pattern to follow here otherwise things get unsafe, so I'd rather have a mention of the deadlock somewhere.